docs/compute.rst

```diff
@@ -0,0 +1,83 @@
+force :math:`F_\rho`
+
+The full code is below:
+::
+
+    from desc.data_index import register_compute_fun
+
+    @register_compute_fun(
+        name="F_rho",
+        label="F_{\\rho}",
+        units="N \\cdot m^{-2}",
+        units_long="Newtons / square meter",
+        description="Covariant radial component of force balance error",
+        dim=1,
+        params=[],
+        transforms={},
+        profiles=[],
+        coordinates="rtz",
+        data=["p_r", "sqrt(g)", "B^theta", "B^zeta", "J^theta", "J^zeta"],
+    )
+    def _F_rho(params, transforms, profiles, data, **kwargs):
+        data["F_rho"] = -data["p_r"] + data["sqrt(g)"] * (
+            data["B^zeta"] * data["J^theta"] - data["B^theta"] * data["J^zeta"]
+        )
+        return data
+
+The decorator ``register_compute_fun`` tells DESC that the quantity exists and contains
+metadata about the quanity. The necessary fields are detailed below:
+
+
+* ``name``: A short, meaningfull name that is used elsewhere in the code to refer to the
+  quanity. This name will appear in the ``data`` dictionary returned by ``Equilibrium.compute``,
+  and is also the argument passed to ``compute`` to calculate the quanity. IE,
+  ``Equilibrium.compute("F_rho")`` will return a dictionary containing ``F_rho`` as well
+  as all the intermediate quantities needed to compute it. General conventions are that
+  covariant components of a vector are called ``X_rho`` etc, contravariant components
+  ``X^rho`` etc, and derivatives by a single character subscript, ``X_r`` etc for :math:`\partial_{\rho} X`
+* ``label``: a LaTeX style label used for plotting.
+* ``units``: SI units of the quanity in LaTeX format
+* ``units_long``: SI units of the quanity, spelled out
+* ``description``: A short description of the quantity
+* ``dim``: Dimensionality of the quantity. Vectors (such as magnetic field) have ``dim=3``,
+  local scalar quantities (such as vector components, pressure, volume element, etc)
+  have ``dim=1``, global scalars (such as total volume, aspect ratio, etc) have ``dim=0``
+* ``params``: list of strings of ``Equilibrium`` parameters needed to compute the quanity
+  such as ``R_lmn``, ``Z_lmn`` etc. These will be passed into the compute function as a
+  dictionary in the ``params`` argument. For most quantities, this will be an empty list.
+* ``transforms``: a dictionary of what ``transform`` objects are needed, with keys being the
+  quantity being transformed (``R``, ``p``, etc) and the values are a list of derivative
+  orders needed in ``rho``, ``theta``, ``zeta``. IE, if the quanity requires
+  :math:`R_{\rho}{\zeta}{\zeta}`, then ``transforms`` should be ``{"R": [[1, 0, 2]]}``
+  indicating a first derivative in ``rho`` and a second derivative in ``zeta``. For most
+  quantites this will be an empty dictionary.
+* ``profiles``: List of string of ``Profile`` quantities needed, such as ``pressure`` or
+  ``iota``. For most quantites this will be an empty list.
```

Comment:
Would be good to clarify that these lists are only for direct dependencies, which is why they end up being empty for most quantities.